packages/kolibri-common/components/SyncSchedule/EditDeviceSyncSchedule.vue

```diff
@@ -0,0 +1,395 @@
+          { label: this.$tr('everyWeek'), value: 86400 },
+          { label: this.$tr('everyTwoWeeks'), value: 86400 },
+          { label: this.$tr('everyMonth'), value: 2592000 },
+        ];
+      },
+      getDays() {
+        const today = new Date();
+        const daysOfWeek = [];
+        const date = new Date(
+          today.getFullYear(),
+          today.getMonth(),
+          today.getDate() + (7 - today.getDay())
+        );
+        for (let i = 0; i < 7; i++) {
+          daysOfWeek.push({ label: this.$formatDate(date, { weekday: 'long' }), value: i });
+          date.setDate(date.getDate() + 1);
+        }
+        return daysOfWeek;
+      },
+
+      SyncTime() {
+        const endTime = new Date();
+        endTime.setHours(24, 0, 0, 0);
+        const interval = 30;
+
+        const times = [];
+        const time = new Date();
+        time.setHours(0, 0, 0, 0);
+
+        while (time < endTime) {
+          times.push(this.$formatTime(time));
+
+          time.setMinutes(time.getMinutes() + interval);
+        }
+
+        return times;
+      },
+    },
+    beforeMount() {
+      this.fetchDevice();
+    },
+    beforeDestroy() {
+      clearInterval(this.timer);
+    },
+    methods: {
+      removeDevice() {
+        this.removeDeviceModal = true;
+      },
+      closeModal() {
+        this.removeDeviceModal = false;
+      },
+      handleDeleteDevice() {
+        this.removeDeviceModal = false;
+        NetworkLocationResource.deleteModel({ id: this.$route.query.id })
+          .then(() => {
+            this.showSnackbarNotification('deviceRemove');
+            history.back();
+          })
+          .catch(() => {
+            this.showSnackbarNotification('deviceNotRemove');
+          });
+      },
+      handleSaveSchedule() {
+        FacilityResource.fetchModel({ id: this.$store.getters.activeFacilityId, force: true }).then(
+          facility => {
+            this.facility = { ...facility };
+            TaskResource.startTask({
+              type: TaskTypes.SYNCPEERFULL,
+              facility: this.facility.id,
+              device_id: this.device.id,
+              baseurl: this.baseurl,
+              enqueue_args: { enqueue_at: this.serverTime, repeat_interval: 2, repeat: 2 },
+            })
+              .then(() => {
+                history.back();
+                this.showSnackbarNotification('syncAdded');
+              })
+              .catch(() => {
+                this.createTaskFailedSnackbar();
+              });
+          }
+        );
+      },
+
+      cancelBtn() {
+        this.$router.push({ name: PageNames.ManageSyncSchedule });
+      },
+      fetchDevice() {
+        NetworkLocationResource.fetchModel({ id: this.$route.query.id }).then(device => {
+          this.device = device;
+          TaskResource.list({ queue: 'facility_task' }).then(tasks => {
+            this.tasks = tasks.filter(device.extra_metadata.device_id === this.id);
```

Comment:
Sorry, I totally missed this previously. The `filter` method of an Array takes a callback function that is called with each element of the Array in turn, so this would then become:
```
this.tasks = tasks.filter(task => task.extra_metadata.device_id === device.id);
```
So for each task, you are looking at its extra_metadata and querying to see which one matches the current device_id.
Another gotcha here is that we probably also need to filter by the facility_id in case we have setup syncing to the same device for multiple facilities!

So it would be something like this:
```
this.tasks = tasks.filter(task => task.extra_metadata.device_id === device.id && task.facility_id === this.$store.getters.activeFacilityId);
```